### ISSUE TITLE
[v2] tags to doc same as tags to blog

### ISSUE THREAD

**Issue by isaac-philip** [facebook/docusaurus#3434]
# 🚀 Feature

Request for copying feature of `/tags` from `/blog/tags` towards `/doc/tags`

-----------

Basically require a way to categorize the docs in groups and refer them in sidebar and also on the uri path of `/doc/tags` similar as how we have for `/blog/tags`

--------

thanks to @slorber for guidance to create this request

**Comment by isaac-philip** [comment]
I can volunteer but new to the project code.

can @slorber or anyone give this issue a starting point where I need to look to understand existing tags functionality.

it can help anyone willing to contribute to this issue.

thanks!

**Comment by amimas** [comment]
I would really love to see tagging capability in the docs. This has been one of the top requested feature from my docs site's tech writers and ux team.

I imagine we could use the front-matter data at the top to contain an array of values that will be considered as "tags" for the corresponding page.

Having the this tagging feature will open other opportunities in addition to what is mentioned above. For example: allow us to be able to filter search results based on tags. Algolia search has the capability to do that. What's missing in Docusaurus currently is a way to be able to tag individual pages.

**Comment by Mac-Kett** [comment]
I want to volunteer if you do this

**Comment by fanny** [comment]
If anyone have interest in contribute I can help you.

You'll need to create a new component in the theme for docs tags: https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-theme-classic/src/theme/BlogTagsListPage/index.tsx

The logic of the route `tags` [here](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-content-blog/src/index.ts#L151). You'll probably need an equivalent in `docusaurus-content-docs`

**Comment by slorber** [comment]
Hi

Will be back from holidays on Monday 28.

We have 2 volunteers so I suggest the first one work on it 😉 I guess the
idea is to make it consistent with the blog comp and try to share some
theme code.

**Comment by slorber** [comment]
Hi,

@isaac-philip are you working on the tags feature? If not @Mac-Kett seems interested to work on it.

**Comment by isaac-philip** [comment]
sorry didn't see the latest messages till being mentioned now.

please proceed as I see people have messaged to guide and take up 😊

sounds great, thanks!

**Comment by slorber** [comment]
Thanks,

@Mac-Kett still interested to work on this?

**Comment by JuanVqz** [comment]
It seems like both volunteers won’t answer, can i take this one? It looks so challenging for me but I think I’ll do it with your help.

**Comment by slorber** [comment]
Yes @JuanVqz thanks ;)

**Comment by JuanVqz** [comment]
does anyone can help me guide me on how to create the route I saw how the [blog tags](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-content-blog/src/index.ts#L150) create the route as @fanny pointed up but the [doc](https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-content-docs/src/index.ts#L44) file it's kind of different. (I'm assuming this is the correct file)

**Comment by slorber** [comment]
@JuanVqz this is how the plugin creates the tags page: https://github.com/facebook/docusaurus/blob/master/packages/docusaurus-plugin-content-blog/src/index.ts#L324

**Comment by JuanVqz** [comment]
maybe I need more experience to do this, I'll let another person do this task

**Comment by slorber** [comment]
Thanks for trying @JuanVqz  :)

**Comment by isaac-philip** [comment]
I will take a try on this today.

**Comment by slorber** [comment]
@isaac-philip are you still interested to work on this? because we might want to implement this soon and can assign this to someone else otherwise

**Comment by isaac-philip** [comment]
```
lerna ERR! yarn run build exited 1 in '@docusaurus/theme-classic'
lerna ERR! yarn run build stdout:
$ tsc --noEmit && yarn babel && yarn prettier
src/theme/DocTagsListPage/index.tsx(12,1): error TS6133: 'Props' is declared but its value is never read.
src/theme/DocTagsListPage/index.tsx(20,33): error TS2709: Cannot use namespace 'Props' as a type.
info Visit https://yarnpkg.com/en/docs/cli/
```
As  per the commit I have added the code to plugin and theme,
errors mean can't go forward to run and verify.

the plugin-content of the docs I copied similar from the one for the blog and would need some refactor too, 
specifically what is the alternative to `blogPost` as I have kept `docTag`

**Comment by slorber** [comment]
@isaac-philip I can't help much if there is no PR. 

If you open a PR (even draft), I can see the real code, the actual netlify error, I can comment on the PR, and might as well commit to your branch to fix issues (if you grant me permission, which is the case by default)

**Comment by isaac-philip** [comment]
@slorber done the needful.
Thanks for the help again!

**Comment by sacha-l** [comment]
Hi @isaac-philip - any updates on this feature? :)

**Comment by isaac-philip** [comment]
Hello @sacha-l , apologies have been away for quite sometime.

Give me till tomorrow to get back to this with some commits else we can pass it to anyone interested.

**Comment by slorber** [comment]
@isaac-philip do you mind if I assign this to @nam-hle ?

**Comment by isaac-philip** [comment]
just updating that have pushed a commit, still WIP but should be ready for a review soon!

took some time to get my head around the architecture of docusaurus & functionality, so thanks for the patience! 

couldn't do without the earlier commits of slorber!

**Comment by lex111** [comment]
This feature was implemented in #3646.

**Comment by waldyrious** [comment]
> This feature was implemented in https://github.com/facebook/docusaurus/pull/3646.

For those who, like me, come across this thread while looking for info on this feature, the documentation is here: https://docusaurus.io/docs/create-doc#doc-tags
